### PR TITLE
Reduce type character size of SceneTableImpl types

### LINKED ISSUE #27111: [Scenes] Cleanup of SceneTableImpl types
As mentioned in https://github.com/project-chip/connectedhomeip/pull/27089/files#r1220198788,

The SceneTableImpl could use some "using" to reduce the size of types such as :
`app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type>`

### PR DESCRIPTION
**Pull request by mpbreton**
Fixes https://github.com/project-chip/connectedhomeip/issues/27111

Add using declaration inside class DefaultSceneHandlerImpl in order to reduce type character size of SceneTableImpl types.

### CODE REVIEW

File: src/app/clusters/scenes-server/SceneTableImpl.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ static constexpr uint16_t kMaxScenesPerFabric   = CHIP_CONFIG_SCENES_MAX_PER_FAB
 static constexpr uint16_t kMaxScenesPerEndpoint = CHIP_CONFIG_MAX_SCENES_PER_ENDPOINT;
 
 using clusterId = chip::ClusterId;
+using namespace app::Clusters::Scenes::Structs;
```
Comment (lpbeliveau-silabs) [inline src/app/clusters/scenes-server/SceneTableImpl.h:34]: Please move all usings into the class to avoid namespace pollution.

Comment (lpbeliveau-silabs) [inline src/app/clusters/scenes-server/SceneTableImpl.h:34]: we do not mean to use a namespace here, see https://github.com/project-chip/connectedhomeip/issues/27111#issuecomment-1579571878 in the issue.

please use:
using AttributeValuePair = app::Clusters::Scenes::Structs::AttributeValuePair::Type;
using List = app::DataModel::List;
etc,
where you see fit.

File: src/app/clusters/scenes-server/SceneTableImpl.cpp [hunk 2]
```diff
@@ -68,15 +67,15 @@ DefaultSceneHandlerImpl::SerializeAdd(EndpointId endpoint,
...
     }
     ReturnErrorOnFailure(pair_iterator.GetStatus());
-    app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> attributeValueList(aVPairs, pairCount);
+    app::DataModel::List<AttributeValuePair::Type> attributeValueList(aVPairs, pairCount);
 
     return EncodeAttributeValueList(attributeValueList, serializedBytes);
 }
 
 CHIP_ERROR DefaultSceneHandlerImpl::Deserialize(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
                                                 app::Clusters::Scenes::Structs::ExtensionFieldSet::Type & extensionFieldSet)
```
Comment (lpbeliveau-silabs) [inline src/app/clusters/scenes-server/SceneTableImpl.cpp:76]: this is another example where a using ExtensionFieldSet = ...
Would be beneficial.

Note, only add using to the .h if it is beneficial to have it there (some types might only be in the .cpp file)

File: src/app/clusters/scenes-server/SceneTableImpl.cpp [hunk 3]
```diff
@@ -23,9 +23,8 @@ namespace chip {
 namespace scenes {
 
 CHIP_ERROR
-DefaultSceneHandlerImpl::EncodeAttributeValueList(
-    const app::DataModel::List<app::Clusters::Scenes::Structs::AttributeValuePair::Type> & aVlist,
-    MutableByteSpan & serializedBytes)
+DefaultSceneHandlerImpl::EncodeAttributeValueList(const app::DataModel::List<AttributeValuePair::Type> & aVlist,
```
Comment (lpbeliveau-silabs) [inline src/app/clusters/scenes-server/SceneTableImpl.cpp:26]: Please use: using List = ... for this type.

**Review by lpbeliveau — CHANGES_REQUESTED** [review]

**Review by andy31415 — CHANGES_REQUESTED** [review]
Please update the PR description and replace the `!!!!!!!!!! Please delete the instructions below and replace with PR description ....` text

File: src/app/clusters/scenes-server/SceneTableImpl.h (hunk 1 above)
Comment (bzbarsky-apple) [inline src/app/clusters/scenes-server/SceneTableImpl.h:34]: Please do not put `using` of any sort at toplevel in headers.  Doing that within class scope is fine, of course.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/scenes-server/SceneTableImpl.h (hunk 1 above)
Comment (mpbreton-silabs) [reply src/app/clusters/scenes-server/SceneTableImpl.h]: Got it thanks for the clarification.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by lpbeliveau — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**